packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -86,6 +87,16 @@ function getScopedDecoratorData(
   if (languageScope === undefined || typeof languageScope === "string") {
     const scope = languageScope ?? context.emitterName;
     if (Object.keys(retval).includes(scope)) return retval[scope];
+
+    // if the scope is negated, we should return undefined
+    const negationScopes = retval[negationScopesKey];
+    if (negationScopes !== undefined && negationScopes.includes(scope)) {
+      return undefined;
+    }
+    // if the scope is not negated, we should return the value for AllScopes
+    else {
+      return retval[AllScopes];
+    }
```

Comment:
do we need this else?

Reply:
No, we don't. It can just fall back to the default return.